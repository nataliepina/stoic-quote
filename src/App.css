@import url('https://fonts.googleapis.com/css2?family=Sora:wght@300;500;600;700&display=swap');

*, 
*::before,
*::after {
  box-sizing: border-box;
}

html {
  scroll-behavior: smooth;
}

:root {
  --font-fam: 'Sora', sans-serif;

  --light: hsl(180, 7%, 97%);
  --medium: hsl(200, 0%, 48%);
  --dark: hsl(210, 100%, 0%);
  --accent: hsl(202, 66%, 25%);
    
  --h1: 3rem;
  --h2: 2.25rem;
  --h3: 1.25rem;
  --body: 1rem;
}

.wrapper {
  width: 100vw;
  height: 100vh;
  margin: 0;
  line-height: 1.6;
  display: grid;
  grid-template-rows: 1fr auto;
}

body {
  font-family: var(--font-fam);
  background: var(--dark);
  color: var(--light);
}

section {
  padding: 3em 3em 6em;
  display: block;
}

footer {
  grid-row-start: 2;
  grid-row-end: 3;
  padding: 3px 6px;
  font-size: 11px;
}

.image-container img {
  width: 320px;
  height: auto;
  filter: grayscale(90%);
  z-index: -2;
}

.image-text {
  writing-mode: vertical-rl;
  text-orientation: mixed;
  font-size: 52px;
  font-weight: bold;
  position: absolute;
  opacity: 50%;
  right: 23px;
  top: 137px;
  z-index: -1;
  letter-spacing: 8px;
}

.quote-container {
  display: block;
  padding: 0 2em;
}

.quote-body {
  font-size: 3.5vw;
  font-weight: 700;
  margin-bottom: 0;
}

.quote-author {
  font-size: var(--h3);
  display: flex;
  flex-direction: row-reverse;
  justify-content: space-between;
  margin-right: 1em;
  align-items: center;
}

.btn {
  width: auto;
  height: auto;
  padding: .5em .7em;
  cursor: pointer;
  background: none;
  transition: .5s;
  font-family: var(--font-fam);
  font-size: var(--body);
  color: var(--light);
  border: none;
}

.btn:hover {
  border: none;
  color: white;
  font-size: 20px;
  transform:scale(1.3);
  transform: rotate(360deg);
  transition: all 0.3s ease-in-out 0s;
}

.refresh {
  margin: 0 2px;
  width: 30px;
  height: 30px;
}

@media (min-width: 600px) {
  section {
    display: flex;
  }
}