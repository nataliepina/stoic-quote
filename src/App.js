import React from 'react';
import './App.css';

const { quotes } = require('./mockDatabase.json')

class App extends React.Component {
  state = {
    quote: {}
  }

  componentDidMount() {
    this.setQuote();
  }
  
  setQuote = () => {
    const quote = quotes[Math.floor(Math.random() * quotes.length)];
    this.setState({ quote });
  }

  render() {
    const { quote, author } = this.state.quote;
    return (  
      <div className='wrapper'>
        <section>
          <p className='quote-body'>{quote}</p>
          <p className='quote-author'>{`- ${author}`}</p>
          <button className='quote-btn' onClick={() => this.setQuote()}>Click</button>
        </section>
        <footer>
            <span className="copyright">Natalie Pina &copy;2020. </span>
        </footer>
      </div>
    )
  }
}

export default App;
