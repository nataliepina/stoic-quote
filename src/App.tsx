import React from 'react';
import './App.css';
import Footer from './components/Footer';
import StoicQuote from './components/StoicQuote';

const App: React.FC = () => {
  const title: string = 'Stoic Quotes';
  return (
    <div className="app">
      <h1 className="title">{title}</h1>
      <StoicQuote />
      <Footer />
    </div>
  );
};

export default App;
