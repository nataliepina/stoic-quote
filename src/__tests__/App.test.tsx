import { render } from '@testing-library/react';
import React from 'react';
import App from '../App';

describe('App tests', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render the title', () => {
    const { getByText } = render(<App />);
    expect(getByText('Stoic Quotes')).toBeInTheDocument();
  });
});
