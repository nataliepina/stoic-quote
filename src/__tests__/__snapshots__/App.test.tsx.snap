// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App tests should match snapshot 1`] = `
<DocumentFragment>
  <main
    class="sc-iqHYGH kVjmEB"
  >
    <div
      class="sc-eCssSg ifsWGG"
    >
      <div
        class="sc-jSgupP iynOAx"
      >
        <button
          class="sc-dlfnbm hZZQlN"
        >
          <span
            aria-label="Light Mode"
            role="button"
          />
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-lightbulb fa-w-11 "
            data-icon="lightbulb"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 352 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M96.06 454.35c.01 6.29 1.87 12.45 5.36 17.69l17.09 25.69a31.99 31.99 0 0 0 26.64 14.28h61.71a31.99 31.99 0 0 0 26.64-14.28l17.09-25.69a31.989 31.989 0 0 0 5.36-17.69l.04-38.35H96.01l.05 38.35zM0 176c0 44.37 16.45 84.85 43.56 115.78 16.52 18.85 42.36 58.23 52.21 91.45.04.26.07.52.11.78h160.24c.04-.26.07-.51.11-.78 9.85-33.22 35.69-72.6 52.21-91.45C335.55 260.85 352 220.37 352 176 352 78.61 272.91-.3 175.45 0 73.44.31 0 82.97 0 176zm176-80c-44.11 0-80 35.89-80 80 0 8.84-7.16 16-16 16s-16-7.16-16-16c0-61.76 50.24-112 112-112 8.84 0 16 7.16 16 16s-7.16 16-16 16z"
              fill="currentColor"
            />
          </svg>
        </button>
        <button
          class="sc-dlfnbm hZZQlN"
        >
          <span
            aria-label="Dark Mode"
            role="button"
          />
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-cloud-moon fa-w-18 "
            data-icon="cloud-moon"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M342.8 352.7c5.7-9.6 9.2-20.7 9.2-32.7 0-35.3-28.7-64-64-64-17.2 0-32.8 6.9-44.3 17.9-16.3-29.6-47.5-49.9-83.7-49.9-53 0-96 43-96 96 0 2 .5 3.8.6 5.7C27.1 338.8 0 374.1 0 416c0 53 43 96 96 96h240c44.2 0 80-35.8 80-80 0-41.9-32.3-75.8-73.2-79.3zm222.5-54.3c-93.1 17.7-178.5-53.7-178.5-147.7 0-54.2 29-104 76.1-130.8 7.3-4.1 5.4-15.1-2.8-16.7C448.4 1.1 436.7 0 425 0 319.1 0 233.1 85.9 233.1 192c0 8.5.7 16.8 1.8 25 5.9 4.3 11.6 8.9 16.7 14.2 11.4-4.7 23.7-7.2 36.4-7.2 52.9 0 96 43.1 96 96 0 3.6-.2 7.2-.6 10.7 23.6 10.8 42.4 29.5 53.5 52.6 54.4-3.4 103.7-29.3 137.1-70.4 5.3-6.5-.5-16.1-8.7-14.5z"
              fill="currentColor"
            />
          </svg>
        </button>
      </div>
      <h1
        class="sc-hKgILt iXBlLH"
      >
        Stoic Quotes
      </h1>
    </div>
    <div
      class="sc-jrAGrp fHoouc"
    >
      <section>
        <div
          aria-busy="true"
          class=""
        >
          <svg
            aria-label="audio-loading"
            fill="lightgray"
            height="80"
            viewBox="0 0 120 30"
            width="80"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="15"
              cy="15"
              r="15"
            >
              <animate
                attributeName="r"
                begin="0s"
                calcMode="linear"
                dur="0.8s"
                from="15"
                repeatCount="indefinite"
                to="15"
                values="15;9;15"
              />
              <animate
                attributeName="fillOpacity"
                begin="0s"
                calcMode="linear"
                dur="0.8s"
                from="1"
                repeatCount="indefinite"
                to="1"
                values="1;.5;1"
              />
            </circle>
            <circle
              attributeName="fillOpacity"
              cx="60"
              cy="15"
              from="1"
              r="9"
              to="0.3"
            >
              <animate
                attributeName="r"
                begin="0s"
                calcMode="linear"
                dur="0.8s"
                from="9"
                repeatCount="indefinite"
                to="9"
                values="9;15;9"
              />
              <animate
                attributeName="fillOpacity"
                begin="0s"
                calcMode="linear"
                dur="0.8s"
                from="0.5"
                repeatCount="indefinite"
                to="0.5"
                values=".5;1;.5"
              />
            </circle>
            <circle
              cx="105"
              cy="15"
              r="15"
            >
              <animate
                attributeName="r"
                begin="0s"
                calcMode="linear"
                dur="0.8s"
                from="15"
                repeatCount="indefinite"
                to="15"
                values="15;9;15"
              />
              <animate
                attributeName="fillOpacity"
                begin="0s"
                calcMode="linear"
                dur="0.8s"
                from="1"
                repeatCount="indefinite"
                to="1"
                values="1;.5;1"
              />
            </circle>
          </svg>
        </div>
      </section>
      <div
        class="sc-kEjbxe dytAeD"
      >
        <button
          class="sc-dlfnbm hZZQlN"
        >
          <span
            class="btn-txt"
          >
            New Quote
          </span>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-redo fa-w-16 "
            data-icon="redo"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M500.33 0h-47.41a12 12 0 0 0-12 12.57l4 82.76A247.42 247.42 0 0 0 256 8C119.34 8 7.9 119.53 8 256.19 8.1 393.07 119.1 504 256 504a247.1 247.1 0 0 0 166.18-63.91 12 12 0 0 0 .48-17.43l-34-34a12 12 0 0 0-16.38-.55A176 176 0 1 1 402.1 157.8l-101.53-4.87a12 12 0 0 0-12.57 12v47.41a12 12 0 0 0 12 12h200.33a12 12 0 0 0 12-12V12a12 12 0 0 0-12-12z"
              fill="currentColor"
            />
          </svg>
        </button>
        <button
          class="sc-dlfnbm hZZQlN"
        >
          <span
            class="btn-txt"
          >
            Copy
          </span>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-copy fa-w-14 "
            data-icon="copy"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M320 448v40c0 13.255-10.745 24-24 24H24c-13.255 0-24-10.745-24-24V120c0-13.255 10.745-24 24-24h72v296c0 30.879 25.121 56 56 56h168zm0-344V0H152c-13.255 0-24 10.745-24 24v368c0 13.255 10.745 24 24 24h272c13.255 0 24-10.745 24-24V128H344c-13.2 0-24-10.8-24-24zm120.971-31.029L375.029 7.029A24 24 0 0 0 358.059 0H352v96h96v-6.059a24 24 0 0 0-7.029-16.97z"
              fill="currentColor"
            />
          </svg>
        </button>
        <button
          class="sc-dlfnbm hZZQlN"
        >
          <span
            class="btn-txt"
          >
            Tweet
          </span>
          <a
            href="https://twitter.com/intent/tweet?text="
            id="tweet-quote"
            rel="noopener noreferrer"
            target="_blank"
            title="Tweet This Quote!"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-twitter fa-w-16 "
              data-icon="twitter"
              data-prefix="fab"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
                fill="currentColor"
              />
            </svg>
          </a>
        </button>
      </div>
    </div>
    <footer
      class="sc-bdfBwQ dCTWBZ"
    >
      <p
        class="sc-gsTCUz eafuQB"
      >
        <a
          data-testid="footer"
          href="https://github.com/nataliepina/stoic-quote"
          rel="noopener noreferrer"
          target="_blank"
        >
          Natalie Pina Â© 2021
        </a>
      </p>
    </footer>
  </main>
</DocumentFragment>
`;
